app/models/adventure.rb

```diff
@@ -1,9 +1,6 @@
 class Adventure < ActiveRecord::Base
-  geocoded_by :address
-  after_validation :geocode  # auto-fetch coordinates
-  # reverse_geocoded_by :latitude, :longitude, address: :address
-  # after_validation :reverse_geocode  # auto-fetch address
-  # after_validation :geocode, if: ->(obj){ obj.address.present? and obj.address_changed? }
+  geocoded_by :address   # can also be an IP address
+  after_validation :geocode, :if => lambda{ |obj| obj.address_changed? }
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Use the new lambda literal syntax `->(params) {...}`.<br>Space missing to the left of {.